tests/ws_integration.rs

```diff
@@ -235,6 +236,155 @@ mod ws_integration {
 		Ok(())
 	}
 
+	#[test(tokio::test)]
+	async fn variable_auth_live_query() -> Result<(), Box<dyn std::error::Error>> {
+		let (addr, _server) = common::start_server_with_defaults().await.unwrap();
+		let socket = &mut common::connect_ws(&addr).await?;
+
+		//
+		// Prepare the connection
+		//
+		let res = common::ws_signin(socket, USER, PASS, None, None, None).await;
+		assert!(res.is_ok(), "result: {:?}", res);
+		let res = common::ws_use(socket, Some("N"), Some("D")).await;
+		assert!(res.is_ok(), "result: {:?}", res);
+
+		// Setup scope
+		let res = common::ws_query(socket, r#"
+        DEFINE SCOPE scope SESSION 2s
+            SIGNUP ( CREATE user SET email = $email, pass = crypto::argon2::generate($pass) )
+            SIGNIN ( SELECT * FROM user WHERE email = $email AND crypto::argon2::compare(pass, $pass) )
+        ;"#).await;
+		assert!(res.is_ok(), "result: {:?}", res);
+
+		// Signup
+		let res = common::ws_send_msg_and_wait_response(
+			socket,
+			serde_json::to_string(&json!({
+				"id": "1",
+				"method": "signup",
+				"params": [{
+					"ns": "N",
+					"db": "D",
+					"sc": "scope",
+					"email": "email@email.com",
+					"pass": "pass",
+				}],
+			}))
+			.unwrap(),
+		)
+		.await;
+		assert!(res.is_ok(), "result: {:?}", res);
+
+		// Sign in
+		let res = common::ws_send_msg_and_wait_response(
+			socket,
+			serde_json::to_string(&json!({
+				"id": "1",
+				"method": "signin",
+				"params": [{
+					"ns": "N",
+					"db": "D",
+					"sc": "scope",
+					"email": "email@email.com",
+					"pass": "pass",
+				}],
+			}))
+			.unwrap(),
+		)
+		.await;
+		assert!(res.is_ok(), "result: {:?}", res);
+		let res = res.unwrap();
+		assert!(res.is_object(), "result: {:?}", res);
+		let res = res.as_object().unwrap();
+
+		// Verify response contains no error
+		assert!(res.keys().all(|k| ["id", "result"].contains(&k.as_str())), "result: {:?}", res);
+
+		// Verify it returns a token
+		assert!(res["result"].is_string(), "result: {:?}", res);
+		let res = res["result"].as_str().unwrap();
+		assert!(res.starts_with("eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzUxMiJ9"), "result: {}", res);
+
+		// Start Live Query
+		let table_name = "test_tableBB4B0A788C7E46E798720AEF938CBCF6";
+		let _live_query_response = common::ws_send_msg_and_wait_response(
+			socket,
+			serde_json::to_string(&json!({
+					"id": "66BB05C8-EF4B-4338-BCCD-8F8A19223CB1",
+					"method": "live",
+					"params": [
+						table_name
+					],
+			}))
+			.unwrap(),
+		)
+		.await
+		.unwrap_or_else(|e| panic!("Error sending message: {}", e))
+		.as_object()
+		.unwrap_or_else(|| panic!("Expected object, got {:?}", res));
+
+		// Wait 2 seconds for auth to expire
+		tokio::time::sleep(tokio::time::Duration::from_secs(2)).await;
+
+		// Start second connection
+		let socket2 = &mut common::connect_ws(&addr).await?;
+
+		// Signin
+		let res = common::ws_send_msg_and_wait_response(
```

Comment:
same as above, let's use `ws_signin`

Reply:
I am a bit confused how that change broke the test. The requests seem correct?
```
2023-08-24T09:25:31.821240Z DEBUG ws_integration::common: Sending message: {"id":"4bc5e1aa2db548e5975610d8bb2f07d0","method":"signin","params":[{"user":"email@email.com","pass":"pass","ns":"N","db":"D","sc":"scope"}]}
2023-08-24T09:25:31.821331Z DEBUG ws_integration::common: Message sent in 91.401µs
2023-08-24T09:25:31.821345Z DEBUG ws_integration::common: Waiting for response...
2023-08-24T09:25:31.825474Z DEBUG ws_integration::common: Response received in 4.117733ms
2023-08-24T09:25:31.825558Z DEBUG ws_integration::common: ws_query result json=Object {"id": String("4bc5e1aa2db548e59[756](https://github.com/surrealdb/surrealdb/actions/runs/5961847412/job/16171867508?pr=2307#step:7:757)10d8bb2f07d0"), "method": String("signin"), "params": Array [Object {"user": String("email@email.com"), "pass": String("pass"), "ns": String("N"), "db": String("D"), "sc": String("scope")}]} msg=Object {"error": Object {"code": Number(-32000), "message": String("There was a problem with authentication")}, "id": String("4bc5e1aa2db548e5975610d8bb2f07d0")}
thread 'ws_integration::variable_auth_live_query' panicked at 'result: Err("unexpected error from query request: Some(Object {\"code\": Number(-32000), \"message\": String(\"There was a problem with authentication\")})")', tests/ws_integration.rs:273:9
```

Reply:
Solved by changing `email@email.com` to `user`. All auth is against `user` now.